### ISSUE TITLE
Tables not loading in Ask a Question > Custom > Select A Table (When Sample Dataset is deleted)

### ISSUE THREAD

**Issue by bentotroop** [metabase/metabase#8532]
I'm using Metabase 0.30.3, application DB and the data source are both in Postgres 10 in separate server.
Testing is using Chrome.

Currently I'm having issue with the Select a table drill down GUI for when navigating from Ask a Question > Custom. The drill down just show my database schemas. Clicking on the schema menu item does nothing, when it's supposed to show secondary menu of tables belonging to the schema.

I can use native query feature perfectly, thus this is not a connection issue.
Also, if I navigate to the page differently, i.e. selecting "View these _table_name_" from any question generated by X-Ray, the resulting page works fine i.e. the Select a table menu widget enlists all schemas and their tables.

Thus, this only happen when navigating from Ask a Question > Custom > Select a table

EDIT: This error no longer happen after I restore Sample Dataset. Tried repeatedly, convinced deleting Sample Dataset is related to the problem.

Console error below:
```
Uncaught (in promise) TypeError: Cannot read property 'name' of null
    at TablePicker (DataSelector.jsx:768)
    at ReactCompositeComponentWrapper._constructComponentWithoutOwner (ReactCompositeComponent.js:306)
    at ReactCompositeComponentWrapper._constructComponent (ReactCompositeComponent.js:282)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:185)
    at Object.mountComponent (ReactReconciler.js:43)
    at Object.updateChildren (ReactChildReconciler.js:119)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
```

![image](https://user-images.githubusercontent.com/42776319/45598001-4631b400-ba07-11e8-9d27-fd8ba51f4027.png)

Please help, keen to upgrade.

**Comment by jornh** [comment]
Related: https://github.com/metabase/metabase/issues/7997

**Comment by paulrosenzweig** [comment]
Resolved by #10766
